### PR TITLE
add missing features to existing resources

### LINKED ISSUE #30: [Enhancement] Add missing fields / endpoints to Images + Servers
* Images have Actions
* Servers have metrics and can change it's rDNS

### PR DESCRIPTION
**Pull request by aufziehvogel**
Implementing the missing features of existing resources (new actions, new attributes) in this PR.

List of requirements compiled from #30 

Some server attributes have already been implemented before this PR.

(!) = Blocked by other PR

- [x] Images have Actions
- [x] Servers have metrics
- [x] Servers Action add_to_placement_group
- [x] Servers Action change_alias_ips
- [x] Servers Action change_dns_ptr
- [x] Servers Action remove_from_placement_group
- [x] Server attribute backup_window
- [x] Server attribute iso (resource type)
- [x] Server attribute labels (currently hash)
- [x] (!) #55 Server attribute load_balancers (currently array) (resource type)
- [x] Server attribute placement_group (resource type)
- [x] Server attribute private_net (currently array) (resource type)
- [x] Server attribute protection ({"delete"=>false, "rebuild"=>false})
- [x] Server attribute public_net (currently hash), has sub fields that are resource types
  - [x] primary IPs
  - [x] floating IPs
  - [x] firewalls (this one is a bit harder to handle as next to the ID it has the additional field `status`, which describes the relationship between the server and the firewall)
    - [x] open issue: the ID is in a hash in the `:id` key, so the parsed firewall object also is in the `:id` key
- [x] Server attribute volumes (currently list) (resource type)
- [x] Network Action change_ip_range
- [x] Floating IPs parameter name on create
- [x] Floating IPs parameter name on update
- [x] Image parameter status on get all
- [x] Image parameter include_deprecated on get all
- [x] Placement groups parameter label_selector on get all
- [x] Server parameter automount on create
- [x] Server parameter firewalls on create
- [x] Server parameter placement_group on create
- [x] Server parameter public_net on create
- [x] Server parameter volumes on create
- [x] SSH key parameter fingerprint on get all
- [x] Volume parameter status on get all

TODO after #55 merged:

- [x] add `it_supports_search` to load balancer spec
- [x] add `it_supports_search` to load balancer type spec

Closes #30

### CODE REVIEW

**Comment by aufziehvogel** [comment]
Solved the problematic `firewalls: [{ id: 12345, status: "applied" }]` with a `raw_data` on our Future object.

Accessing the Firewall object with `server.public_net[:firewalls][0][:id] # type Firewall` would be weird. A user can expect to get the firewall on `server.public_net[:firewalls][0]`. But then we lose the `applied` status information.

Luckily, `server.public_net[:firewalls][0]` does not really hold a `Firewall` object, it holds a `Future` which transparently forwards every method to a `Firewall` object. I added a method `raw_data` to this `Future`, which returns the actual Hash/JSON data.

Users can now request the hidden data (e.g. `applied` status) like this:

```
# basically a Firewall object (in reality a Future, but forwards everything to Firewall)
server.public_net[:firewalls][0]

# get the { id: 12345, status: 'applied' } hash
server.public_net[:firewalls][0].raw_data
```

**Review by tonobo — APPROVED** [review]
